.github/workflows/build.yml

```
@@ -0,0 +1,91 @@
+name: Build
+on:
+  push:
+    branches: [master, dev, branch-*, dogfood-*]
+  pull_request:
+  merge_group:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 22 * * 1'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
```

I'd prefer not to hardcode main branch
```suggestion
  cancel-in-progress: ${{ github.ref_name != github.event.repository.default_branch }}
```